### PR TITLE
[WIP] better overall user experience

### PR DESCRIPTION
**Pull request by pocesar**
* better errors
* country selector
* sessionPool mark bad for repeatedly failed selectors
* support for all requestListSources
* check urls before adding them to the requests
* some JSdocs and code typos
* linting (long way to go)

### CODE REVIEW

File: INPUT_SCHEMA.json [hunk 1]
```diff
@@ -24,11 +24,164 @@
...
+                "th",
+                "lo",
+                "my",
+                "km",
+                "ko",
+                "ja",
+                "zh-CN",
+                "zh-TW"
+            ],
+            "enumTitles": [
```
Comment (metalwarrior665) [inline INPUT_SCHEMA.json:105]: Looks a bit weird that some are lowercase and some uppercase, is that correct?

File: INPUT_SCHEMA.json [hunk 2]
```diff
@@ -204,9 +357,13 @@
             "type": "boolean",
             "description": "Cache places location between runs in named KeyVal. Check location in polygon before enqueuing place to requestQueue.",
             "default": false
+        },
+        "forceEng": {
```
Comment (metalwarrior665) [inline INPUT_SCHEMA.json:361]: We can remove this, just make sure it is backward compatible in the code (we will deprecate it over time)

File: src/main.js [hunk 3]
```diff
@@ -134,30 +143,45 @@ Apify.main(async () => {
...
+                '--disable-features=IsolateOrigins,site-per-process',
+            ],
         },
         maxOpenPagesPerInstance: maxPagesPerBrowser,
         // Not sure why this is here
         retireInstanceAfterRequestCount: 100,
     };
 
-    const proxyConfiguration = await Apify.createProxyConfiguration(proxyConfig);
+    const proxyConfiguration = await Apify.createProxyConfiguration({ ...proxyConfig });
```
Comment (metalwarrior665) [inline src/main.js:164]: Why this change? This would crash if proxyConfig is null.

File: src/places_crawler.js [hunk 4]
```diff
@@ -392,11 +426,22 @@ const setUpCrawler = (crawlerOptions, scrapingOptions, stats, allPlaces) => {
                     urlPatterns: ['/maps/vt/', '/earth/BulkMetadata/', 'googleusercontent.com'],
                 });
             }
-            if (forceEng) request.url += '&hl=en';
+            const mapUrl = new URL(request.url);
+
+            if (forceEng) {
```
Comment (metalwarrior665) [inline src/places_crawler.js:431]: Again, we can remove this now.

File: INPUT_SCHEMA.json (hunk 1 above)
Comment (pocesar) [reply INPUT_SCHEMA.json:105]: yes, took directly from the language selector from their page

File: INPUT_SCHEMA.json (hunk 2 above)
Comment (pocesar) [reply INPUT_SCHEMA.json:361]: ok

File: src/main.js (hunk 3 above)
Comment (pocesar) [reply src/main.js:164]: why would the proxyConfig be null? it's not marked as nullable in INPUT_SCHEMA, and it's by default, `undefined`, and when you spread `undefined` you get an empty object

File: src/places_crawler.js (hunk 4 above)
Comment (pocesar) [reply src/places_crawler.js:431]: there's a caveat though: right now, if you pass a `?hl=cs` url it will be converted to en. if you pick another language from the language selector other than `cs` in this case, it will overwrite that as well. this can be confusing, since it defaults to `en`
